### DISCUSSION TITLE
Ignore foreign key in column with Prisma migration

### DISCUSSION THREAD

**Discussion by danielbuechele** [prisma/prisma#17606] [Q&A]
I use `prisma db pull/push` to manage some, but not all of my tables in Postgres. However, some of my tables managed by through Prisma have foreign keys to tables not managed by Prisma. I marked these columns with `@ignore`, but when running `prisma db push` I'm still getting an error that the table to which my foreign key is pointing isn't in my schema.
Is there a way for prisma migrations to just entirely ignore this column?

**Comment by nurul3101** [comment]
Hello @danielbuechele 👋 

We have a Feature Request for this use case:
- https://github.com/prisma/prisma/issues/8864

For now, I would recommend you just define the tables that are not managed by Prisma in the schema file and mark them as [@@ignore](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#ignore-1). Additionally, can you have a look at this [workaround](https://github.com/prisma/prisma/issues/8864#issuecomment-1301362357) and see if it works for you?

Once the feature request is implemented, this use case would be much easier to achieve. Also, if you could add a 👍 to the feature request, then it would help our product team in prioritising it.

**Reply by danielbuechele** [reply]
Thanks for the quick reply. What you linked is certainly related and I encountered this issue, too. I solved it by moving my tables which are not managed by Prisma into a different schema. The approach with a second user mentioned as a workaround is also a possible solution here.
However, just hiding the tables from Prisma is not enough for my use-case, because tables which are actually managed by Prisma, have foreign keys to tables not managed by Prisma. What I would need is a way for Prisma to ignore these columns entirely. `@ignore` still causes Prisma to fail, because the ignored column has a foreign key to a table not managed by Prisma.

**Reply by nurul3101** [reply]
Thanks for sharing the solution that worked for you. Moving tables to a different schema is indeed smart 🙌 

Do you mind adding a comment to the GitHub Issue that @ignore doesn't work for you as ignored column has a foreign key to the table that is not managed by Prisma? Our engineering team can consider this when designing a solution.

**Comment by vasily-polonsky** [comment]
Hi @danielbuechele 

could you pls share you solution?
If I have schemaA with tableA, and schemaB with tableB, and foreign key tableA.propId->tableB.id, I am getting an error below when do `db pull` or `migrate dev`

```
Error: P4002

The schema of the introspected database was inconsistent: The schema of the introspected database was inconsistent: Cross schema references are only allowed when the target schema is listed in the schemas property of your datasource. `schemaA.tableA` points to `schemaB.tableB` in constraint `tableB_prop_id_fkey`. Please add `tableB` to your `schemas` property and run this command again.
```

If I create user with access only to schemaA, and create manually foreign key, error stays the same. Really curious about your solution

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
